modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/DbaseFileTest.java

```diff
@@ -186,31 +188,79 @@ public void testEmptyFields() throws Exception {
+        // A null string should result in padding
+        String formattedString = formatter.getFieldString(sizeInBytes, null);
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("        ", formattedString);
 
-        assertEquals("          ".getBytes().length, formattedString.getBytes().length);
+        // A short string will be padded
+        formattedString = formatter.getFieldString(sizeInBytes, "cat");
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("cat     ", formattedString);
 
-        // test when the string is too big.
-        stringWithInternationChars = '\u20ac' + "1234567890";
-        formattedString = formatter.getFieldString(10, stringWithInternationChars);
+        // A string that has the right number of bytes needs no padding
+        formattedString = formatter.getFieldString(sizeInBytes, "12345678");
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("12345678", formattedString);
 
-        assertEquals("          ".getBytes().length, formattedString.getBytes().length);
+        // larger strings get trucated
+        formattedString = formatter.getFieldString(sizeInBytes, "12345678910");
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("12345678", formattedString);
     }
 
     @Test
     public void testUTF8Chars() throws Exception {
-        // looks like "A, B, C" in an editor, but the B and C are actually weird UTF-8 entities
         DbaseFileWriter.FieldFormatter formatter =
                 new DbaseFileWriter.FieldFormatter(
                         StandardCharsets.UTF_8, TimeZone.getDefault(), false);
-        String test = "A, \u0412, \u0421";
+
+        int sizeInBytes = 4;
+
+        // a short string will be padded
+        String formattedString = formatter.getFieldString(sizeInBytes, "\u0412");
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("\u0412  ", formattedString);
+
+        // a string of size btyes need no padding
+        formattedString = formatter.getFieldString(sizeInBytes, "\u0412\u0412");
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("\u0412\u0412", formattedString);
+
+        // large strings get trucated
+        formattedString = formatter.getFieldString(sizeInBytes, "\u0412\u0412\u0412");
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("\u0412\u0412", formattedString);
+
+        // if a multi-byte character is the last to be removed then padding may be required to
+        formattedString = formatter.getFieldString(sizeInBytes, "\u0412A\u0412\u0412");
+        assertEquals(sizeInBytes, formattedString.getBytes().length);
+        assertEquals("\u0412A ", formattedString);
+    }
+
+    @Test(timeout = 10000)
```

Comment:
Please increase the timeout. The GitHub PR checks work in virtualized environments whose performance is highly unpredictable, while we want to avoid a borderline stuck build, we cannot demand a test to "go fast".

Reply:
I'll remove